subprojects/build-cache/src/main/java/org/gradle/caching/configuration/internal/DefaultBuildCacheConfiguration.java

```diff
@@ -57,18 +58,24 @@ public BuildCache getLocal() {
 
     @Override
     public <T extends BuildCache> T local(Class<T> type) {
+        if (!type.equals(DirectoryBuildCache.class)) {
```

Comment:
This check is not necessary, since the method `local(Class, Action)` will be called which then emits the nagging anyway.

Reply:
Good point.